uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/calendar-feeds/link-modal/calendar-feeds-link-modal.service.js

```diff
@@ -17,6 +17,12 @@ define(function () {
      * @param {String} url - the url to display on the modal.
      */
     function open (url) {
+      var appendToElement = $document.find('#bootstrap-theme');
```

Comment:
I thought modals are styled properly outside namespaces, am I wrong?

Anyway, if we absolutely must do that, there should be a comment to the function I think:

```js
/**
 * Opens a modal with the calendar feed link.
 * @NOTE it injects the modal inside the #bootstrap-theme, if exists, otherwise inside <body>
 *
 * @param {String} url - the url to display on the modal.
 */
```

Reply:
does this work in the SSP? There is no `#bootstrap-theme` element in it

Reply:
@igorpavlov @AkA84 modals are not displayed properly outside `#bootstrap-theme`, they look like a random block of text/content that appears at the top of the page.

They look OK in other admin sections such as Leave Calendar or Tasks because those sections import bootstrap through SCSS or they are including `bootstrap.css`. They also look OK in SSP because Bootstrap is globally available through SSP Theme.

In this case, instead of importing Bootstrap so it's globally available I decided to attach the modal to `#bootstrap-theme` element when available.

Reply:
i agree that the documentation should explicitly reference what's going on with `#bootstrap-theme` and more importantly _why_